src/components/apps/AppStepper.js

```diff
@@ -128,22 +129,25 @@ const AppStepper = React.forwardRef((props, ref) => {
 
     return (
         <Stepper ref={ref} alternativeLabel nonLinear activeStep={activeStep}>
-            {steps.map((step, index) => (
-                <Step key={step.label}>
-                    <StepButton
-                        id={buildID(
-                            baseId,
-                            ids.APP_STEPPER.STEP_BTN,
-                            index + 1
-                        )}
-                        onClick={handleStep(index)}
-                    >
-                        <StepLabel error={stepError(index)}>
-                            {step.label}
-                        </StepLabel>
-                    </StepButton>
-                </Step>
-            ))}
+            {steps.map((step, index) => {
+                const completed = stepCompleted && stepCompleted(index);
+                const hasError = stepError && stepError(index);
+
+                return (
+                    <Step key={step.label} completed={completed}>
```

Comment:
Pretty simple change on the app `Step` side to get it to display as completed, though the state management in the form side is a little more complicated; such as managing step `touched` state, so it's not instantly marked as completed or with an error until the user actually does something with the step.